lang/java/reef-io/src/main/java/org/apache/reef/io/watcher/EventType.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.reef.io.watcher;
+
+/**
+ * Event types that Watcher subscribes.
+ */
+public enum EventType {
```

Comment:
As per the discussion we had on the list, it might make sense to reverse the words in these names, e.g. `TaskFailed` instead of `FailedTask`. This applies similarly to other parts of this PR (schemas, ...)